dealership/contracts.py

```diff
@@ -1,12 +1,49 @@
+from dealership.vehicles import Car, Truck, Motorcycle
+
 class Contract(object):
     pass
 
-
 class BuyContract(Contract):
     def __init__(self, vehicle, customer, monthly_payments):
-        pass
-
+        self.vehicle = vehicle
+        self.customer = customer
+        self.monthly_payments = monthly_payments
+        
+    def interest_rate(self):
+        interest_dict = {Car : 1.07, Motorcycle: 1.03, Truck : 1.11}
+        return (interest_dict[type(self.vehicle)] * self.monthly_payments * self.vehicle.sale_price() / 100)
+    
+    def total_value(self):
+        price = self.vehicle.sale_price() + self.interest_rate() 
+        if not self.customer.is_employee():
+            return price
+        else:
```

Comment:
You could eliminate these 3 lines of code with just "return price - (price * 0.1) 

 else:
             discount = price * 0.1
             return price - discount